### ISSUE TITLE
Add option to deployAll to wait until deployment is succeeded

### ISSUE THREAD

**Issue by christianhuening** [qaware/gradle-cloud-deployer#50]
Hi,

is there any way to make the deployAll task run and block until the deployed Pods are up and running? In my case we're trying to use this in a Jenkinsfile as part of an integration test suite and we're running in the problem, that the deployment task finishes too soon and thus the tests fail. 

cheers
christian

**Comment by sjahreis** [comment]
Hi christian,

Right now there is no blocking feature like that. The plugin blocks until the pods are or the deployment is created, but not until the pods are completely running.

It should be possible to implement such a feature using healthcheck adresses. We'll discuss that! I'll keep you informed!

Until that, there is no other solution than implementing such a logic in your test cases. You could use a rest client which blocks until a request is successful...

Best regards,
Simon

**Comment by christianhuening** [comment]
Yeah that's what we did. The other feature would be quite nice though, I think. Thanks for your effort!

**Comment by sjahreis** [comment]
Hi christian,

we decided to add this feature to the further roadmap (See #45). It will be added soon!

Thanks for your suggestion!

Best regards,
Simon

**Comment by sjahreis** [comment]
For Kubernetes we could use this REST endpoint:
http://kubernetes.io/kubernetes/third_party/swagger-ui/#!/api%2Fv1/readNamespacedComponentStatus

**Comment by sjahreis** [comment]
For Marathon we could use this REST endpoint:
https://mesosphere.github.io/marathon/docs/generated/api.html#v2_apps__app_id__get
